client-src/elements/chromedash-survey-questions.ts

```diff
@@ -0,0 +1,177 @@
+import {LitElement, TemplateResult, css, html, nothing} from 'lit';
+import {customElement, property, state} from 'lit/decorators.js';
+import {SHARED_STYLES} from '../css/shared-css.js';
+import {SlChangeEvent, SlInput} from '@shoelace-style/shoelace';
+import {Feature, StageDict, User} from '../js-src/cs-client';
+import {GateDict} from './chromedash-gate-chip';
+import {GATE_TYPES} from './form-field-enums.js';
+import {GATE_QUESTIONNAIRES} from './gate-details.js';
+import {autolink} from './utils.js';
+
+@customElement('chromedash-survey-questions')
+export class ChromedashSurveyQuestions extends LitElement {
+  static get styles() {
+    return [
+      ...SHARED_STYLES,
+      css`
+        #questionnaire {
+          padding: var(--content-padding-half);
+          border-radius: var(--border-radius);
+          background: var(--table-alternate-background);
+        }
+        #questionnaire * + * {
+          padding-top: var(--content-padding);
+        }
+        #questionnaire ul {
+          padding-left: 1em;
+        }
+        #questionnaire ul li {
+          list-style: disc;
+        }
+        #questionnaire ol li {
+          list-style: auto;
+          margin-left: var(--content-padding);
+        }
+        .question {
+          padding: var(--content-padding-half);
+        }
+        .instructions {
+          padding: var(--content-padding-half);
+          margin-bottom: var(--content-padding-large);
+        }
+      `,
+    ];
+  }
+
+  @state()
+  feature!: Feature;
+  @state()
+  gate!: GateDict;
+  @state()
+  loading = true;
+
+  refetch() {
+    window.csClient.getGates(this.feature.id).then(resp => {
+      for (const g of resp.gates) {
+        if (g.id === this.gate.id) {
+          this.gate = g;
+          break;
+        }
+      }
+    });
+  }
+
+  renderQuestionnaireSkeleton(): TemplateResult {
+    return html`
+      <!-- prettier-ignore -->
+      <div id="questionnaire">Loading...</div>
+      <p class="instructions">&nbsp;</p>
+    `;
+  }
+
+  handleFieldChange(name: string, value: string | boolean) {
+    window.csClient
+      .updateGate(this.feature.id, this.gate.id, null, {[name]: value})
+      .then(() => {
+        this.refetch();
+      });
```

Comment:
This will send a request for every time any field changes state, including for each new character change in an input field. Would a "Submit" option be a better choice to avoid the potential of redundant requests?

Reply:
Actually, the sl-change event is only sent once the user un-focuses on the input field or navigates away.  So, it is not one for every keystroke.  I am thinking that automatic updates are better than having a submit button because there are already enough buttons to compete for the user's attention.  Specifically, after filling in these answers, the user will be often be clicking on the "Request review" button.  I don't want them to have to fill-in and press submit before being able to press the "Request review" button.

Reply:
Ah, you are correct about the unfocus aspect 🙂  - although, maybe there is a risk of the user not unfocusing for something - it's like you said about the likelihood of the "Request review" button being the next action - so that sounds fine to me